backend/src/main/java/ai/giskard/web/socket/WorkerStatusSocketService.java

```diff
@@ -18,19 +17,22 @@ public class WorkerStatusSocketService {
     private final MLWorkerService mlWorkerService;
     private final SimpMessagingTemplate simpMessagingTemplate;
 
-    @PostConstruct
-    public void init() {
-        Timer timer = new Timer();
-        timer.schedule(new TimerTask() {
-            @Override
-            public void run() {
-                Map<String, Boolean> data = new HashMap<>();
-                data.put("connected", mlWorkerService.isExternalWorkerConnected());
+    @EventListener
+    public void handleSessionSubscribeEvent(SessionSubscribeEvent event) {
+        sendCurrentStatus();
+    }
+
+    @EventListener
+    public void handleUpdateWorkerStatusEvent(UpdateWorkerStatusEvent event) {
+        sendCurrentStatus();
+    }
+
+    public void sendCurrentStatus() {
+        Map<String, Boolean> data = new HashMap<>();
+        data.put("connected", mlWorkerService.isExternalWorkerConnected());
```

Comment:
I think that we should take more advantage of the `UpdateWorkerStatusEvent`.

For me the event should carry a variable `boolean connected` and  this variable should be set as a global variable in `WorkerStatusSocketService` by default at `false`. This way we won't need to call `isExternalWorkerConnected` that will then call the MLWorker since we already have the status.

Actually `isExternalWorkerConnected` would ideally check for the latest status event

Reply:
I'm not sure about changing `isExternalWorkerConnected`. I think it's important to keep this method forcing to check if there is a worker connected at the moment it's called instead of replicating the status of the last emitted `UpdateWorkerStatusEvent`.